File: Workshop/JavaBasicsProject/src/Basics/OO/Counter/Counter.java
package Basics.OO.Counter;

//  Create Counter class
//  which has a counter (integer) field
//  at creation it should have a default value 0 or can be specified by a parameter
//  we can add another whole number to this counter: add(number)
//  we can just increase the counter's value by one: add() (no parameters)
//  we can get() the current counter value
//  and we can reset() the counter to the initial value


public class Counter {
  Integer counter = 0;
  Integer initialValue = counter;

  public Counter() {
  }

  public Counter(Integer counter) {
    this.counter = counter;
    this.initialValue = counter;
  }

  public void add() {
    this.counter++;
  }

  public void add(int number) {
    this.counter += number;
  }

  public Integer get() {
    return counter;
  }

  public void reset() {
    counter = initialValue;
  }
}

File: Foundation/week_02-week_04/PythonBasicsProject/week_03/oop/counter/counter.py
# # Counter
#
# - Create `Counter` class
class Counter():

    #   - which has a `counter` (integer) field
    #   - at creation it should have a default value `0` or can be specified
    #   by a parameter
    def __init__(self, counter=0):
        self.counter = counter
        self.init_value = counter

    #   - we can add another whole number to this counter: `add(number)`
    #   - we can just increase the counter's value by one: `add()` (no parameters)
    def add(self, number=1):
        self.counter += number

    #   - we can `get()` the current counter value
    def get(self):
        return self.counter

    #   - and we can `reset()` the counter to the initial value
    def reset(self):
        self.counter = self.init_value
# - Check if everything is working fine with the proper test
#   - Download `test_counter.py` and place it next to your solution
#   - Run the test file as a usual python program
